MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,95 @@ public class ResumeButton: UIControl {
     }
 }
 
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIView {
+    
+    private static let textInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 6)
+    
+    lazy var label: SpeedLimitLabel = .forAutoLayout()
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed\nLimit\n", comment: "Speed limit sign main text.")
+    let paragraphStyle = NSMutableParagraphStyle()
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            let fontSize = UIFont.systemFontSize
+            
+            let strings = NSMutableAttributedString()
+            
+            strings.append(NSMutableAttributedString(string: defaultText.uppercased(), attributes: [
+                .paragraphStyle: paragraphStyle,
+                .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: String(describing: speedLimit.speed), attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2),
+                .paragraphStyle: paragraphStyle
+                ]))
+            
+            strings.append(NSMutableAttributedString(string: "\n\(speedLimit.speedUnits.localizedSpeedUnit)", attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize / 1.3),
+                .paragraphStyle: paragraphStyle
+                ]))
+            
+            label.attributedText = strings
+        }
+    }
+    
+    @objc dynamic public var borderColor: UIColor? {
+        get {
+            guard let color = layer.borderColor else { return nil }
+            return UIColor(cgColor: color)
+        }
+        set {
+            layer.borderColor = newValue?.cgColor
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    
+    func commonInit() {
+        addSubview(label)
+        paragraphStyle.lineHeightMultiple = 0.9
+        paragraphStyle.alignment = .center
+        layoutMargins = SpeedLimitSign.textInsets
+        label.lineBreakMode = .byWordWrapping
+        label.numberOfLines = 4
+        label.textAlignment = .center
+        label.pinInSuperview(respectingMargins: true)
+    }
+    
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+        layer.cornerRadius = bounds.midY / 8
+    }
+}
+
+extension SpeedUnit {
+    var localizedSpeedUnit: String {
+        switch self {
+        case .kilometersPerHour:
+            return NSLocalizedString("KILOMETERS_PER_HOUR", bundle: .mapboxNavigation, value: "kph", comment: "Inform the user on limits for speed.")
```

Comment:
“kph” is a relatively rare abbreviation for kilometers per hour on signage. “km/h” is much more common.